app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -85,6 +86,116 @@ def registry_authorize
       return false
     end
 
+    def get_blob_push_props(path_string=nil)
+      # format should follow one of these formats:
+      #   - /v2/{org_label}/{product_label}/{name}/blobs/uploads...
+      #   - /v2/id/{org_id}/{product_id}/{name}/blobs/uploads...
+      #   - /v2/{org_label}/{product_label}/{name}/manifests/...
+      #   - /v2/id/{org_id}/{product_id}/{name}/manifests/...
+      # inputs not matching format will return {valid_format: false}
+      path_string = @_request.fullpath if path_string.nil?
+      segments = path_string.split('/')
+
+      if segments.length >= 7 && segments[0] == "" && segments[1] == "v2" && 
+        segments[2] != "id" && (segments[5] == "blobs" || segments[5] == "manifests")
+
+        return {
+          valid_format: true,
+          schema: "label",
+          organization: segments[2],
+          product: segments[3],
+          name: segments[4]
+        }
+      elsif segments.length >= 8 && segments[0] == "" && segments[1] == "v2" && 
+        segments[2] == "id" && (segments[6] == "blobs" || segments[6] == "manifests")
+
+        return {
+          valid_format: true,
+          schema: "id",
+          organization: segments[3],
+          product: segments[4],
+          name: segments[5]
+        }
+      else
+        return {valid_format: false}
+      end      
+    end
+
+    def check_blob_push_props(props=nil)
+      props = get_blob_push_props if props.nil?
+
+      # check basic url field syntax
+      unless props[:valid_format]
+        render_error('custom_error', :status => :unprocessable_entity,
+          :locals => { :message => "Invalid format. Container pushes should follow '/organization_label/product_label/name' OR '/organization_id/product_id/name' schema." })
```

Comment:
Try to be consistent with rocket syntax and default to colon syntax, I think colon syntax is now the norm in Ruby. Rocket is necessary sometimes still, like when using strings instead of labels.

Reply:
Done, thanks!